### PR TITLE
Add command showDocs to vscode plugin

### PR DESCRIPTION
**Pull request by roonyh**
## Purpose
This command can be used to view markdown documentations in a bal file

image

### CODE REVIEW

File: composer/documentation/src/index.jsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './components/App';
+
+export default function render(ast, el) {
+    console.log(el)
```
Comment (jeradrutnam) [inline composer/documentation/src/index.jsx]: Shall we remove the `console.log(el)` ?

File: tool-plugins/vscode/src/docs/activator.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+import { workspace, commands, window, Uri, ViewColumn, ExtensionContext, TextEditor, WebviewPanel, TextDocumentChangeEvent } from 'vscode';
```
Comment (jeradrutnam) [inline tool-plugins/vscode/src/docs/activator.ts]: Shall we make it 2 lines? I think this exceeds 120 character limit. :)

File: tool-plugins/vscode/package.json [hunk 3]
```diff
@@ -150,6 +150,10 @@
             {
                 "command": "ballerina.showExamples",
                 "title": "Ballerina: Show Examples"
+            },
+            {
+                "command": "ballerina.showDocs",
+                "title": "Ballerina: Show Docs"
```
Comment (hevayo) [inline tool-plugins/vscode/package.json]: Add Category instead of Ballerina:

File: tool-plugins/vscode/src/docs/renderer.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { Uri, ExtensionContext } from 'vscode';
+import * as path from 'path';
+
+export function render(context: ExtensionContext): string {
+    let composerJsSrc;
+    if (process.env.COMPOSER_DEBUG === "true") {
```
Comment (hevayo) [inline tool-plugins/vscode/src/docs/renderer.ts]: The condition should change to debug false. and try to make this a common component where the script section can be contributed.

File: composer/documentation/src/components/App.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import Documentation from './Documentation';
+
+export default class App extends React.Component {
```
[the remark this answers is not in the document]
Comment (roonyh) [reply composer/documentation/src/components/App.jsx]: Changed in the follow up commit

File: composer/extended-language-client/src/extended-language-client.ts [hunk 6]
```diff
@@ -100,6 +100,15 @@ export class ExtendedLangClient extends LanguageClient {
         return this.sendRequest("ballerinaExample/list", args);
     }
 
+    fetchDocs(uri: Uri): Thenable<any> {
```
[the remark this answers is not in the document]
Comment (roonyh) [reply composer/extended-language-client/src/extended-language-client.ts]: Removed

File: tool-plugins/vscode/src/docs/renderer.ts (hunk 4 above)
Comment (roonyh) [reply tool-plugins/vscode/src/docs/renderer.ts]: Changed the condition. Seems the existing util function has the ability to have additional scripts contributed. But I've used an onload function as well. Shall we do the refactoring to use a common component later. We have to consider how chunking might affect this as well.

File: tool-plugins/vscode/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (roonyh) [reply tool-plugins/vscode/package.json]: Renamed

Comment (roonyh) [reply tool-plugins/vscode/package.json]: Added category Ballerina

File: tool-plugins/vscode/src/docs/renderer.ts (hunk 4 above)
Comment (hevayo) [reply tool-plugins/vscode/src/docs/renderer.ts]: +1

**Pull request merged**